python/lsst/ctrl/bps/wms/panda/cmd_line_embedder.py

```diff
@@ -0,0 +1,94 @@
+import os
+
+
+class CommandLineEmbedder:
+    """
+    Class embeds static (constant across a task) values into the pipeline execution command line
+    and resolves submission side environment variables
+
+    Parameters
+    ----------
+    config : `~lsst.ctrl.bps.BPSConfig`
+            BPS configuration that includes the list of dynamic (uniques per job)
+            and submission side resolved variables
+
+    """
+    def __init__(self, config):
+        self.leave_placeholder_params = config.get("placeholderParams", ['qgraphNodeId', 'qgraphId'])
+        self.submit_side_resolved = config.get("submitSideResolvedParams", ['USER'])
+
+    def replace_static_parameters(self, cmd_line, lazy_vars):
+        """ Substitutes the lazy parameters in the command line which are static,
+        the same for every job in the workflow and could be defined once. This function offloads the edge
+        node processing and number of parameters transferred together with job
+
+
+        Parameters
+        ----------
+        cmd_line: `str` command line to be processed
+        lazy_vars: `dict` of lazy variables and its values
+
+        Returns
+        -------
+        processed command line
+        """
+
+        for param_name, param_val in lazy_vars.items():
+            if param_name not in self.leave_placeholder_params:
+                cmd_line = cmd_line.replace('{'+param_name+'}', param_val)
+        return cmd_line
+
+    def resolve_submission_side_env_vars(self, cmd_line):
+        """ Substitutes the lazy parameters in the command line which are defined and resolved
+         on the submission side
+
+        Parameters
+        ----------
+        cmd_line: `str` command line to be processed
+
+        Returns
+        -------
+        processed command line
+        """
+
+        for param in self.submit_side_resolved:
+            cmd_line = cmd_line.replace('<ENV:'+param+'>', os.getenv(param))

```

Comment:
This will fail if the env var does not exist but it might be preferable to at least catch it and report which environment variable you expected to exist because otherwise `replace() argument 2 must be str, not None` is a bit opaque.